File: app/src/main/java/com/msaggik/algorithmsdatastructures/java/graph/graph_tree/data_structures/Queue.java
package com.msaggik.algorithmsdatastructures.java.graph.graph_tree.data_structures;

public interface Queue<D> {
    void offer(D data); // add data in end queue
    D poll(); // reed and delete data in start queue
    D peek(); // reed data in start queue
    boolean isEmpty();
}

File: app/src/main/java/com/msaggik/algorithmsdatastructures/kotlin/graph/graph_tree/data_structures/Queue.kt
package com.msaggik.algorithmsdatastructures.kotlin.graph.graph_tree.data_structures

interface Queue<D> {
    fun offer(data: D) // add data in end queue
    fun poll(): D // reed and delete data in start queue
    fun peek(): D // reed data in start queue
    val isEmpty: Boolean
}